deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/Schedules.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2016 Pinterest, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *  
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *    
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.pinterest.teletraan.resource;
+
+import com.pinterest.deployservice.bean.ScheduleState;
+import com.pinterest.arcee.handler.ProvisionHandler;
+import com.pinterest.deployservice.bean.EnvironBean;
+import com.pinterest.deployservice.bean.ScheduleBean;
+import com.pinterest.deployservice.dao.ScheduleDAO;
+import com.pinterest.deployservice.dao.EnvironDAO;
+
+import com.pinterest.teletraan.TeletraanServiceContext;
+import com.pinterest.teletraan.exception.TeletaanInternalException;
+import com.pinterest.deployservice.common.CommonUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.swagger.annotations.*;
+
+import javax.validation.Valid;
+import javax.ws.rs.*;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.SecurityContext;
+
+import java.util.Collection;
+import java.util.List;
+
+@Path("/v1/schedules")
+
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class Schedules {
+    private static final Logger LOG = LoggerFactory.getLogger(Schedules.class);
+    private ScheduleDAO scheduleDAO;
+    private EnvironDAO environDAO;
+
+    private ProvisionHandler provisionHandler;
+
+    public Schedules(TeletraanServiceContext context) {
+        scheduleDAO = context.getScheduleDAO();
+        environDAO = context.getEnvironDAO();
+        provisionHandler = new ProvisionHandler(context);
+    }
+
+    @GET
+    @Path("/{scheduleId : [a-zA-Z0-9\\-_]+}")
```

Comment:
Since the update and override APIs take envName and stageName, I would suggest to take envName & stageName here as well.